pkg/scheduler/plugins/predicates/predicates.go

```diff
@@ -402,14 +409,16 @@ func (pp *predicatesPlugin) OnSessionOpen(ssn *framework.Session) {
 			predicateStatus = append(predicateStatus, podsNumStatus)
 		}
 
-		predicateByStablefilter := func(pod *v1.Pod, nodeInfo *k8sframework.NodeInfo) ([]*api.Status, bool, error) {
+		predicateByStablefilter := func(nodeInfo *k8sframework.NodeInfo) ([]*api.Status, bool, error) {
 			// CheckNodeUnschedulable
 			predicateStatus := make([]*api.Status, 0)
 			status := nodeUnscheduleFilter.Filter(context.TODO(), state, task.Pod, nodeInfo)
 			nodeUnscheduleStatus := api.ConvertPredicateStatus(status)
 			if nodeUnscheduleStatus.Code != api.Success {
 				predicateStatus = append(predicateStatus, nodeUnscheduleStatus)
-				return predicateStatus, false, fmt.Errorf("plugin %s predicates failed %s", nodeUnscheduleFilter.Name(), status.Message())
+				if nodeUnscheduleStatus.Code == api.UnschedulableAndUnresolvable {
```

Comment:
I think this code should be encapsulated in a function like `predicateFail(status)` that means the `predicateFn` doesn't need to continue.
Besides, it can also exits immediately when `api.Error/api.Skip/api.Wait` occurred. It is better to call the same function `predicateFail(status)` at **allocate** and **preempt** phase(in function `session.PredicateForAllocateAction` and `session.PredicateForPreemptAction`).

Reply:
The statuses that should terminate the predicate phase have been refactored into a separate function. As for the logic where actions determine pass/fail based on the status set, I believe the current composition approach is a better choice.